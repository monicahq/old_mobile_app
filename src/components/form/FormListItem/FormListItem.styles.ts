import {PixelRatio, StyleSheet} from 'react-native';

export const borderColor = '#b5b5b5';

export const styles = StyleSheet.create({
  container: {
    paddingVertical: 15,
    paddingHorizontal: 15,
    borderColor,
    borderBottomWidth: 1 / PixelRatio.get(),
    width: '100%',
  },
  last: {
    borderBottomWidth: 0,
  },
});
