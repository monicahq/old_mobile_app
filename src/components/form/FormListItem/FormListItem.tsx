import React, {PureComponent} from 'react';
import {View} from 'react-native';

import {styles} from './FormListItem.styles';

interface IFormListItemProps {
  last?: boolean;
}

export class FormListItem extends PureComponent<IFormListItemProps> {
  public static defaultProps = {
    last: false,
  };
  public render() {
    const {children, last} = this.props;

    return (
      <View style={[styles.container, last ? styles.last : null]}>
        {children}
      </View>
    );
  }
}
