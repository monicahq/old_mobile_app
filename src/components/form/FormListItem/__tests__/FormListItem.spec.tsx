import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {View} from 'react-native';
import {FormListItem} from '../FormListItem';

describe('Components', () => {
  describe('FormListItem', () => {
    it('should renders correctly 1', () => {
      expect(toJson(shallow(<FormListItem />))).toMatchSnapshot();
    });
    it('should renders correctly 2', () => {
      expect(toJson(shallow(<FormListItem last={true} />))).toMatchSnapshot();
    });
    it('should renders correctly 3', () => {
      expect(
        toJson(
          shallow(
            <FormListItem>
              <View />
            </FormListItem>
          )
        )
      ).toMatchSnapshot();
    });
    it('should renders correctly 4', () => {
      expect(
        toJson(
          shallow(
            <FormListItem last={true}>
              <View />
            </FormListItem>
          )
        )
      ).toMatchSnapshot();
    });
  });
});
