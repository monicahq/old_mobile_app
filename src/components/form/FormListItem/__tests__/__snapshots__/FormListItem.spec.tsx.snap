// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components FormListItem should renders correctly 1 1`] = `
<ForwardRef(View)
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
        "width": "100%",
      },
      null,
    ]
  }
/>
`;

exports[`Components FormListItem should renders correctly 2 1`] = `
<ForwardRef(View)
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
        "width": "100%",
      },
      Object {
        "borderBottomWidth": 0,
      },
    ]
  }
/>
`;

exports[`Components FormListItem should renders correctly 3 1`] = `
<ForwardRef(View)
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
        "width": "100%",
      },
      null,
    ]
  }
>
  <ForwardRef(View) />
</ForwardRef(View)>
`;

exports[`Components FormListItem should renders correctly 4 1`] = `
<ForwardRef(View)
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#b5b5b5",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
        "width": "100%",
      },
      Object {
        "borderBottomWidth": 0,
      },
    ]
  }
>
  <ForwardRef(View) />
</ForwardRef(View)>
`;
